src/rpc/misc.cpp

```diff
@@ -1430,6 +1431,47 @@ UniValue getsparklatestcoinid(const JSONRPCRequest& request)
     return UniValue(latestCoinId);
 }
 
+UniValue checkifmncollateral(const JSONRPCRequest& request)
+{
+    if (request.fHelp || request.params.size() != 2)
+        throw std::runtime_error(
+                "checkifmncollateral\n"
+                "\nReturns bool value.\n"
+                "\nArguments:\n"
+                "  \"txHash\"\n"
+                "  \"index\"\n"
+                + HelpExampleCli("checkifmncollateral", "\"b476ed2b374bb081ea51d111f68f0136252521214e213d119b8dc67b92f5a390\""  "\"0\" ")
+                + HelpExampleRpc("checkifmncollateral", "\"b476ed2b374bb081ea51d111f68f0136252521214e213d119b8dc67b92f5a390\"" "\"0\" ")
+        );
+
+    std::string strTxId;
+    int index;
+
+    try {
+        strTxId = request.params[0].get_str();
+        index = std::stol(request.params[1].get_str());
+    } catch (std::logic_error const & e) {
+        throw std::runtime_error(std::string("An exception occurred while parsing parameters: ") + e.what());
+    }
+
+    uint256 txid = uint256S(strTxId);
+
+    CTransactionRef tx;
+    uint256 hashBlock;
+    if(!GetTransaction(txid, tx, Params().GetConsensus(), hashBlock, true))
+        throw std::runtime_error("Unable to get the transction");
```

Comment:
typo in "transaction" plus wouldn't it be better to say something like "Unknown transaction"?